mlx/backend/cuda/CMakeLists.txt

```diff
@@ -37,6 +39,24 @@ target_sources(
 
 target_compile_definitions(mlx PRIVATE MLX_USE_CUDA)
 
+# Embed kernel sources in binary for JIT compilation.
+file(
+  GLOB MLX_JIT_SOURCES
+  RELATIVE ${CMAKE_CURRENT_SOURCE_DIR}
+  "${CMAKE_CURRENT_SOURCE_DIR}/kernels/*.h"
+  "${CMAKE_CURRENT_SOURCE_DIR}/kernels/*.cuh")
```

Comment:
This is a bit of a nit, but the directory layout has a confusing naming scheme. It seems like `kernels/` is not actually where the "kernels" go.. but where some of the utilities for the kernels go. 

Can we reconsider the name of this subdirectory? Or alternatively just flatten the directory structure all into `cuda` and explicitly list the sources we want to include here?

Reply:
Yea I like `device` more. That's a great suggestion!